packages/fee-estimation/README.md

```diff
@@ -0,0 +1,394 @@
+
+const data = encodeFunctionData({
+  functionName: 'burn',
+  abi: optimistABI,
+  args: [BigInt('0x77194aa25a06f932c10c0f25090f3046af2c85a6')],
+});
+```
+
+This will return a 0x-prefixed hex string that represents the encoded function data.
+
+## Testing
+
+The package provides a set of tests that you can run to verify its operation. The tests are a great resource for examples. You can find them at `./src/estimateFees.spec.ts`.
+
+## Other methods
+
+This package also provides lower level methods for estimating gas
+
+### getL2Client()
+
+This method returns a Layer 2 (L2) client that communicates with an L2 network.
+
+```ts
+getL2Client(options: ClientOptions): PublicClient;
+```
+
+#### Parameters
+- `options: ClientOptions` - The options required to initialize the L2 client.
+
+#### Returns
+- `PublicClient` - Returns a public client that can interact with the L2 network.
+
+#### Example
+
+```ts
+const clientParams = {
+  chainId: 10,
+  rpcUrl: process.env.VITE_L2_RPC_URL ?? 'https://mainnet.optimism.io',
+} as const;
+
+const client = getL2Client(clientParams);
+```
+
+---
+
+### baseFee()
+
+Returns the base fee.
+
+```ts
+baseFee({ client, ...params }: GasPriceOracleOptions): Promise<bigint>;
+```
+
+#### Parameters
+- `{ client, ...params }: GasPriceOracleOptions` - The options required to fetch the base fee.
+
+#### Returns
+- `Promise<bigint>` - Returns a promise that resolves to the base fee.
+
+#### Example
+
+```ts
+const blockNumber = BigInt(106889079);
+const paramsWithClient = {
+  client: clientParams,
+  blockNumber
+};
+const baseFeeValue = await baseFee(paramsWithClient);
+```
+
+---
+
+### decimals()
+
+Returns the decimals used in the scalar.
+
+```ts
+decimals({ client, ...params }: GasPriceOracleOptions): Promise<bigint>;
+```
+
+#### Parameters
+- `{ client, ...params }: GasPriceOracleOptions` - The options required to fetch the decimals.
+
+#### Returns
+- `Promise<bigint>` - Returns a promise that resolves to the decimals used in the scalar.
+
+#### Example
+
+```ts
+const decimalsValue = await decimals(paramsWithClient);
+```
+
+---
+
+### gasPrice()
+
+Returns the gas price.
+
+```ts
+gasPrice({ client, ...params }: GasPriceOracleOptions): Promise<bigint>;
+```
+
+#### Parameters
+- `{ client, ...params }: GasPriceOracleOptions` - The options required to fetch the gas price.
+
+#### Returns
+- `Promise<bigint>` - Returns a promise that resolves to the gas price.
+
+#### Example
+
+```ts
+const gasPriceValue = await gasPrice(paramsWithClient);
+```
+
+---
+
+### getL1Fee()
+
+Computes the L1 portion of the fee based on the size of the rlp encoded input transaction, the current L1 base fee, and the various dynamic parameters.
+
+```ts
+getL1Fee(data: Bytes, { client, ...params }: GasPriceOracleOptions): Promise<bigint>;
+```
+
+#### Parameters
+- `data: Bytes` - The transaction call data as a 0x-prefixed hex string.
+- `{ client, ...params }: GasPriceOracleOptions` - Optional lock options and provider options.
```

Comment:
I didn't review these well can replace these with more explicit params. This is copy pasted from the tests